src/mono/mono/metadata/monitor.c

```diff
@@ -1340,6 +1341,8 @@ mono_monitor_wait (MonoObjectHandle obj_handle, guint32 ms, MonoBoolean allow_in
 
 	LOCK_DEBUG (g_message ("%s: (%d) Unlocked %p lock %p", __func__, id, obj, mon));
 
+	mono_atomic_inc_i64 (&thread_waits);
```

Comment:
I want the count to be increment only when the thread yields. Not sure if that's the case here.